### DISCUSSION TITLE
Beware Things that are Free(U) - My alternative settings (WIP/Testing)

### DISCUSSION THREAD

**Discussion by leomaxwell973** [lllyasviel/stable-diffusion-webui-forge#540] [General]
Short Version:

FreeU - Reconfigured #s for realism models and generations

Noticed FreeU, by default and when using its Instructions on its github are geared toward anime/cgi/semi-real, this is my attempt at reconfiguring the numbers for a small-moderate realism boost while enjoying the other quality aspects with a realism geared model/project.

### **S2 1.41**
(results may vary, by model, and other factors)
### NO LIFEGUARD ON DUTY

If you want to hear the when, how, and why, as well as notes, and interesting things I noticed while fiddling with numbers, enjoy the text wall below. 

--------------------------------------------------------------------------------------------------------------------------------------------------
Cause:

A lot of my generations are geared for realism, I heard about FreeU briefly, heard it helped with generations. Flipped it on, and forgot about it, but then noticed something, I am having a tough time yanking stuff away from anime style when using MMs, which most, if not all, are trained on semi-real at best, typically anime models. (Some odd reason not too noticeable on normal image gen, odd.)

turned everything off and tested one by one, FreeU was biggest culprit, (K-highres too but only by a fraction of what FreeU was doing.)
Thought all was all well when i checked the github to look into it, set the non-default recommended settings, and noticed things were tad bit better... but still very CGI, couldn't get away from it, tested again, yep FreeU still doing it.

So, Was about to forgeetaboutit, then I thought, if it can make realism -> CGI/Anime, surly I could just flip these numbers and have it work for me after all?

well, yes, but wasn't just a simple flip, not for me at least, maybe I was salty and just wanted it to make-up for wasted time? xD but here  are the settings I've eventually settled on for my current model and how it was turning out results with my first test run of... well every number combo I could stay sane up to xD 

--------------------------------------------------------------------------------------------------------------------------------------------------
Crunching numbers and taking note:

### **S2 1.41**

S1/2 notes: noted some relativity properties between S! and S2, also realized the default introductory page suggested 0.9 and 0.2 which does not allow for a difference this great (1.59) clearly, therefore S1&S2 must be >1 (milage and conclusion may vary by model/ckpt(?) and in my current opinion, with limited knowledge (no lifeguard on duty).);; 
point being, idk if S1=3 is ACTUALLY best, however, while leaving it on 3 while testing S2@1.41 was best in that particular configuration, leaving me to believe its more important to find the optimal difference between the S1/2, rather than any specific value (though im sure some values are better than others but if S1/2 too close/far won't matter, i think), or I could be wrong and got lucky or even chose not the most optimal values; only <insert deity here> knows.

afterthought: with S1, theoretically and likely dealing with foreground attributes, values will likely vary on character distance from "viewer" position. This optimization was narrowed down with an example that was within whispering distance close-up, for reference. and may explain why I felt no need to test further than S1@3 after locking in S2s value.

### **B2 1.98**

B2 and a little of everything notes: interesting observation on B2: highest levels (later found to be ~1.97-2.00) incrementally makes ears, specifically/exclusively, grow longer the higher it is, slightly more pointy, elvish like, subtle amount but was noticeable unlike other subtle changes when changing 0.01 values. B1@ 1.66, S1@3 S2@1.41 at time; 
confirmed lowering down to 1, ears shorter, thicker, rounder; also noted when dropping to 1, more hair strands/messy uncombed hair look. 
Conclusion: B2 ~ties-in/~intersects with the ~Unet/~latent-space/~weights responsible for the attribute values for character head components, and/or other more main-like visual features and attributes, likely.
 
Did not hard research the rest or have the same eureka moment(s) but S1/2 felt like they effected background/foreground as a somewhat whole in a threshold like manner; (S1 being foreground, IIRC)

B1 felt like it controlled more general details of actors/objects, for example skin and other flat textures and similar, B2 affecting shapes and details more complex than a flat plane

another observation: at 1.96 of B2 hairbow barley begins to form &1.97 of B2, a hairbow appeared, stealthy like, blended in with background colorwise almost, @1.99 poofs with no fadeout between like the lower values and it appearing, taking it as a sign to park B2@1.98 since was struggling to judge what's just different or better for the fine tuning of the last bit of B2. 

significance of hairbow, aside from this tiebreaker so to speak,is more observations of B2's association with the more geometric (not flat) details, acknowledgment of its role may prove useful later.

**Comment by slann303** [comment]
Thanks for your experimentations and sharing your results with us. I appreciate that! I have used the recommended values from the github mainly for realist models (realisticvision 5.1, epicphotogasm last unicorn, cyberrealistic 4.1) and derived very good results. I have tested your values and unfortunately the results are "weird". I turned off Hires.fix and Self attention guidance, but the results are the same. I have switched back to the recommended values.
